app/src/main/java/com/android/streetworkapp/model/park/ParkViewModel.kt

```diff
@@ -80,6 +81,18 @@ open class ParkViewModel(private val repository: ParkRepository) : ViewModel() {
    */
   fun createPark(park: Park) = viewModelScope.launch { repository.createPark(park) }
 
+  /**
+   * Get or create a park by its location.
+   *
+   * @param location The park location.
+   */
+  fun getOrCreateParkByLocation(location: ParkLocation) {
+    viewModelScope.launch {
+      val park = repository.getOrCreateParkByLocation(location)
+      _park.value = park
```

Comment:
Comment: I think that having a "Park" and a "CurrentPark" is more confusing than useful. I see why this is implemented in order to distinguish the 2 uses, but we have, at the moment, not really used 2 separate versions. My advice would be to keep only one but I see why you would prefer to keep the 2 separate.

Reply:
Basically I agree with you - right now in our project we only need a current park. However, I'd rather keep potential features for future implementation than remove something already implemented. What do you think of it?

Reply:
I understand your argument. I think we can keep it as it is for the moment.